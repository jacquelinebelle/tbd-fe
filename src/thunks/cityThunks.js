import { gatherCities, setCurrentCity, isLoading, gotError } from '../actions';
import { getCityDetails, getCityImages, getCityScores } from '../api/cityCalls';

export const cityThunk = (city) => {
    return async (dispatch) => {
        try {
            dispatch(isLoading(true));
            let image = await getCityImages(city);
            let scores = await getCityScores(city);
            let cityObject = {...image, ...scores}
            dispatch(isLoading(false));
            console.log(cityObject)
            dispatch(setCurrentCity(cityObject));
            // this.setState({ city: cityObject })
        } catch (error) {
        dispatch(gotError(error.message));
      }
    }
  }

  