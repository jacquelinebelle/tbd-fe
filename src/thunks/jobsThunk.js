import { gatherJobs, isLoading, gotError, resetJobs } from '../actions';
import { getJobs } from '../api/jobCalls';
// import { getCityscores, getCityImages } from '../api/cityCalls';

export const jobsThunk = (params) => {
    return async (dispatch) => {
      try {
        dispatch(resetJobs());
        dispatch(isLoading(true));
        const jobs = await getJobs(params);
        dispatch(isLoading(false));
        dispatch(gatherJobs(jobs));
      } catch (error) {
        dispatch(gotError(error.message));
      }
    }
  }