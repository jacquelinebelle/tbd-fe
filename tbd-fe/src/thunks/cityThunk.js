import { gatherCities, isLoading, gotError } from '../actions';
import { getCityDetails } from '../api/cityCalls';

export const cityThunk = (locations) => {
    return async (dispatch) => {
        try {
            dispatch(isLoading(true));
            const cities = await Promise.all(locations.map(city => {
                console.log(city)
                return getCityDetails(city);
            }));
            dispatch(isLoading(false));
            dispatch(gatherCities(cities));
            return cities;
        } catch (error) {
        dispatch(gotError(error.message));
      }
    }
  }