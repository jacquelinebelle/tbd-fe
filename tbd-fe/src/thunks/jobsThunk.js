import { gatherJobs, isLoading, gotError } from '../actions';
import { getJobs } from '../api/jobCalls';

export const jobsThunk = (params) => {
    return async (dispatch) => {
      try {
        dispatch(isLoading(true));
        const jobs = await getJobs(params);
        dispatch(isLoading(false));
        dispatch(gatherJobs(jobs));
      } catch (error) {
        dispatch(gotError(error.message));
      }
    }
  }