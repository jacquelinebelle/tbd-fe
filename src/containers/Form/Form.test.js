import React from 'react';
import {shallow} from 'enzyme';
import {Form, mapDispatchToProps} from './index';
import { jobsThunk } from '../../thunks/jobsThunk';


describe("Form", () => {
  let wrapper;
  let mockJobsThunk; 

  beforeEach(function() {
    mockJobsThunk = jest.fn()
    wrapper = shallow(
      <Form jobsThunk={mockJobsThunk}/>
    )
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should update state when user types in form fields', () => {
    wrapper.instance().handleChange = jest.fn();
    let newValue = 'Hi';
    wrapper.find("form").simulate("change", {
      target: {name: "keywords", value: newValue}
    })
    expect(wrapper.state().keywords).toEqual(newValue)
  })

  it('should update state when user types in form fields', () => {
    wrapper.instance().handleChange = jest.fn();
    let newValue = '100000';
    wrapper.find("form").simulate("change", {
      target: {name: "salary", value: newValue}
    })
    expect(wrapper.state().salary).toEqual(newValue)
  })

  it('should update state when user types in form fields', () => {
    wrapper.instance().handleChange = jest.fn();
    let newValue = 'Dallas, TX';
    wrapper.find("form").simulate("change", {
      target: {name: "location", value: newValue}
    })
    expect(wrapper.state().location).toEqual(newValue)
  })

  it('should update state when user types in form fields', () => {
    wrapper.instance().handleChange = jest.fn();
    let newValue = 2;
    wrapper.find("form").simulate("change", {
      target: {name: "radius", value: newValue}
    })
    expect(wrapper.state().radius).toEqual(newValue)
  })

  describe('mapDispathToProps', () => {
    it('should call JobThunk when dispatch', () => {
      const mockDispatch = jest.fn();
      const mockState = {keywords: 'developer'}
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.gatherJobs(mockState)
      expect(mockDispatch).toHaveBeenCalled();
    })
  })
})



