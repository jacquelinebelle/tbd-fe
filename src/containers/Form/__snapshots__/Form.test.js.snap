// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match snapshot 1`] = `
<form
  className="search-form"
  onChange={[Function]}
>
  <label
    className="search-label"
  >
    Job Title / Keywords
    <input
      className="form-input keyword-input"
      name="keywords"
      type="text"
    />
  </label>
  <label
    className="search-label"
  >
    Desired Salary
    <input
      className="form-input salary-input"
      name="salary"
      type="text"
    />
  </label>
  <label
    className="search-label"
  >
    Location
    <input
      className="form-input location-input"
      name="location"
      type="text"
    />
  </label>
  <label
    className="search-label radius-label"
  >
    Location Radius
    <input
      className="form-input radius-input"
      name="radius"
      type="number"
    />
  </label>
  <Link
    to="/results"
  >
    <button
      className="form-btn"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#B2D1E4",
        }
      }
    >
      Submit
    </button>
  </Link>
</form>
`;
